proposals/records-wip.md

```diff
@@ -76,3 +76,47 @@ each primary constructor parameter to the corresponding property of the other ty
 ```C#
 override Equals(object o) => Equals(o as T);
 ```
+
+## `with` expression
+
+A `with` expression is a new expression using the following syntax.
+
+```antlr
+with_expression
+    : switch_expression
+    | switch_expression 'with' anonymous_object_initializer
+```
+
+A `with` expression allows for "non-destructive mutation", designed to
+produce a copy of the receiver expression with modifications to properties
+listed in the `anonymous_object_initializer`.
+
+A valid `with` expression has a receiver with a non-void type. The receiver type must contain an accessible instance method called `With` with
+the appropriate parameters and return type. It is an error if there are multiple non-override `With` methods. If there are multiple `With` overrides,
+there must be a non-override `With` method, which is the target method. Otherwise, there must be exactly one `With` method.
+
+On the right hand side of the `with` expression is an `anonymous_object_initializer` with a
+sequence of assignments with a field or property of the receiver on the left-hand side of the
```

Comment:
>a [](start = 29, length = 1)

... with the name of a ...

Reply:
I think that's covered below in

> Each property or field in the right-hand side of the With
expression must also correspond to a parameter of the same name in the `With` method.